python-lib/tagger.py

```diff
@@ -1,38 +1,45 @@
-"""Module to prepare the tagging of the documents
-splits on sentences and instanciates a Matcher to tag the documents"""
+"""Main module to tag the documents"""
 
 from spacy_tokenizer import MultilingualTokenizer
 from dkulib_io_utils import generate_unique
 from spacy.matcher import PhraseMatcher
 from fastcore.utils import store_attr
-from typing import AnyStr
+from typing import AnyStr, List
 import pandas as pd
+from time import perf_counter
+import logging
+from formatter_instanciator import FormatterInstanciator
 
 
 class Tagger:
     def __init__(
         self,
-        text_df,
-        ontology_df,
-        text_column,
-        language,
-        language_column,
-        tag_column,
-        category_column,
-        keyword_column,
-        lemmatize,
-        case_sensitive,
-        normalize,
-        mode,
+        text_df: pd.DataFrame,
```

Comment:
Thought: wouldn't it be more natural to specify text_df, text_column, language_column, as method arguments instead of class attributes?

This way, you can initialize the class on a given ontology, with given text simplification parameters, and then apply to different text dataframes.

Reply:
d646387